canvas/tree.py

```diff
@@ -0,0 +1,111 @@
+def _is_collapsible(tree):
+    """ Checks to see if the children of the node can be collapsed. """
+    if len(tree.children) == 0:
+        return False
+
+    for c in tree.children:
+        if not c.is_tip():
+            return False
+    return True
+
+
+def _trim_level(tree, table):
+    """ Collapses tree leaves together by one level. """
+    collapsed_tree = tree.copy()
```

Comment:
You can extend an object from a library without a problem, which will avoid doing this and also move the code directly to skbio when ready. An example: http://stackoverflow.com/a/2706023